packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,198 @@
+
+  // Only validate (known) classes and aggregations
+  const xmlElement = attribute.parent;
+
+  // UI5 Class case
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  if (ui5Class !== undefined) {
+    if (isValidUI5ClassAttribute(ui5Class, attribute)) {
+      return [];
+    }
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  // UI5 Aggregation case
+  const ui5Aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  // Aggregations don't have additional allowed attributes
+  if (ui5Aggregation !== undefined) {
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  // No validations on unknown tags
+  return [];
+}
+
+function isAttributeNameAlwaysValid(attribute: XMLAttribute): boolean {
+  // This will never happen - it was checked before calling this function
+  /* istanbul ignore if */
+  if (attribute.key === null) {
+    return true;
+  }
+  // xmlns attributes are always valid
+  // TODO "xmlns:" should not return true (change when using new function in xml-tools)
+  if (isXMLNamespaceKey(attribute.key)) {
+    return true;
+  }
+
+  const { ns, name } = splitAttributeByNamespace(attribute);
+
+  // The following attributes are valid on all tags
+  // undefined name means all attribute names in the namespace are valid
+  const ALWAYS_VALID_ATTRIBUTES = [
+    // See: https://sapui5.hana.ondemand.com/#/topic/de9fd55c69af4b46863f5d26b5d796c4
+    { ns: "sap.ui.dt", name: "designtime" },
+    // See: https://sapui5.hana.ondemand.com/#/topic/b11d853a8e784db6b2d210ef57b0f7d7
+    // TODO this attribute is not valid for FragmentDefinition inside xml views
+    { ns: "sap.ui.core", name: "require" },
+    // See: https://sapui5.hana.ondemand.com/#/topic/263f6e5a915f430894ee290040e7e220
+    { ns: TEMPLATING_NS, name: "require" },
+    // See: https://sapui5.hana.ondemand.com/#/topic/91f0c3ee6f4d1014b6dd926db0e91070
+    { ns: CUSTOM_DATA_NS, name: undefined },
+  ];
+
+  if (
+    find(
+      ALWAYS_VALID_ATTRIBUTES,
+      (_) => _.ns === ns && (_.name === undefined || _.name === name)
+    ) !== undefined
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+function isValidUI5ClassAttribute(
+  ui5Class: UI5Class,
+  attribute: XMLAttribute
+): boolean {
+  // The following attributes are valid on all classes
+  // undefined fqn means the attribute is valid on all classes
+  // TODO Should these be in the model? Should they be suggested for code completion?
+  const ALWAYS_VALID_CLASS_ATTRIBUTES = [
```

Comment:
we always search on the `ns` first, would a dictionary be more efficient / appropriate to model this?